prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -219,15 +219,15 @@ static int rx_cm_data(SOCKET fd, struct ofi_ctrl_hdr *hdr,
 	return FI_SUCCESS;
 }
 
-static int tx_cm_data(SOCKET fd, int type, struct poll_fd_info *poll_info)
+static int tx_cm_data(SOCKET fd, uint8_t type, struct poll_fd_info *poll_info)
 {
 	struct ofi_ctrl_hdr hdr;
-	int ret;
+	ssize_t ret;
 
 	memset(&hdr, 0, sizeof(hdr));
 	hdr.version = OFI_CTRL_VERSION;
 	hdr.type = type;
-	hdr.seg_size = htons(poll_info->cm_data_sz);
+	hdr.seg_size = (uint16_t) htons(poll_info->cm_data_sz);
```

Comment:
What is windows doing that it needs this cast?  odd...

Reply:
That is a mistake. It should be
hdr.seg_size = htons((uint16_t) poll_info->cm_data_sz);

Reply:
There are still some warnings related to tcp. I will update this PR again.